reactioncommerce/reaction-next-starterkit
# feat: add segment tracking

Issue: #58 
Impact: **minor**
Type: **feat**

## Issue

Add an event tracking API to the app and have that data be sent to Segment.

## Solution

- Add segment scripts using th next js example https://github.com/zeit/next.js/tree/canary/examples/with-segment-analytics
- Use the NYT react tracking lib https://github.com/NYTimes/react-tracking
- Create a custom dispatch and HOC decorators under `lib/tracking`
- Add a `config/analytics` directory with the segment tracking implementation as well as an example for creating more
- Add an ENV var `SEGMENT_ANALYTICS_WRITE_KEY` to `.env.example` for easy configuration
- Create sample tracking HOC's for tracking a product and product list, to be fully implemented in the future by #56 and #57 
- Added docs to outlining how to set up. use, and customize tracking.
- Add the `_app.js` override component with the root level tracking

## Breaking

- `this.props.url` is now `this.props.router`

## Testing

1. Create a segment account and get the API key
2. Start the app and navigate around
3. In the segment dashboard under `sources > javascript > debugger` see events being tracked
